pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -145,21 +145,17 @@ var labelReconcileInfo = []struct {
 	ensureSecondaryExists bool
 }{
 	{
-		// Reconcile the beta and the stable OS label using the beta label as
-		// the source of truth.
-		// TODO(#73084): switch to using the stable label as the source of
-		// truth in v1.18.
-		primaryKey:            kubeletapis.LabelOS,
-		secondaryKey:          v1.LabelOSStable,
+		// Reconcile the beta and the stable OS label using the stable label as the source of truth.
+		// TODO(#89477): no earlier than 1.22: drop the beta labels if they differ from the GA labels
+		primaryKey:            v1.LabelOSStable,
```

Comment:
Wondering if the primary key switch should have happened one release before the beta label was removed from kubelet. 

I think this can potentially break some upgrade scenarios where kubelet runs on the control plane nodes. For example:
1. some critical daemonset (like CNI) uses node selector `beta.kubernetes.io/os`.
2. first control plane node is upgraded to v1.19 and kubelet only sets the GA label `kubernetes.io/os` now
3. kube-controller-manager is still running on v1.18 and is still expecting `beta.kubernetes.io` to be the source of truth
4. new v1.19 node is stuck in NotReady because it only has the kubernetes.io/os label and v1.18 kube-controller-manager is expecting beta.kubernetes.io/os to be the source of truth.

Reply:
Ideally users should update their node selectors to use kubernetes.io/os though since that label will exist on both v1.18 and v1.19 nodes. The other workaround here is to restart the v1.18 kube-controller-manager pods and until the v1.19 kube-controller-manager is the leader.

Reply:
Fwiw I don't think this technically violates version skew policy since the issue only occurs if you run a v1.19 kubelet when kube-controller-manager is still on v1.18, which is more common when you use kubelet static pods to run the control plane.